### DISCUSSION TITLE
TypeError: 'slice' object isn't callable

### DISCUSSION THREAD

**Discussion by lbuque** [micropython/micropython#12206] [General]
micropython  1.20.0
board: esp32 + 2m psram + 8m flash

I am using uasyncio to make a ui, when _BtnB_wasClicked_event is triggered, it will be passed to App's _keycode_dpad_down_event_handler.

![image](https://github.com/micropython/micropython/assets/46518962/945b9327-3955-4acc-baac-276942573d3c)

![image](https://github.com/micropython/micropython/assets/46518962/2d8cfc74-622c-4bd6-8cb2-26f719b0c079)

However, sometimes it works fine, and sometimes the following exception occurs:

![image](https://github.com/micropython/micropython/assets/46518962/6780e53a-f79a-44fa-805f-6df77a82c31d)

**Comment by peterhinch** [comment]
I have moved this to discussions because you have provided no evidence of e MicroPython bug. To enable us to find the bug in the code you'll need to provide more information including the traceback and the code where the error occurs.

**Comment by sosi-deadeye** [comment]
Without the line number from the traceback it's hard to find the bug.

You're calling somewhere a `slice` object and those are not callable.
You can't create a `slice` object directly in Micropython, but the `__getitem__` method of a class gets a `slice` object if a range is selected.

Example:
```python
class Foo:
    def __getitem__(self, key):
        print(key)

# not a slice object
# Foo()[1]
# 1

# slice object
Foo()[:]
# slice(None, None, None)

# slice object
Foo()[0:]
# slice(0, None, None)


# slice object
Foo()[0:10]
# slice(0, 10, None)

# slice object
Foo()[0:10:2]
# slice(0, 10, 2)
```

Somewhere in your code, a name is assigned to a slice object. Later it's called, which raises a `TypeError`.

**Comment by lbuque** [comment]
I made the following modifications, which seem to avoid the exception:

![image](https://github.com/micropython/micropython/assets/46518962/0679b0a9-6676-4d19-a064-2ab5c90386da)
